### PR TITLE
hostnamed: exit-on-idle tweaks

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

File: src/libsystemd/sd-bus/sd-bus.c [hunk 1]
```diff
@@ -4466,3 +4466,22 @@ _public_ int sd_bus_enqueue_for_read(sd_bus *bus, sd_bus_message *m) {
         bus->rqueue[bus->rqueue_size++] = bus_message_ref_queued(m, bus);
         return 0;
 }
+
+_public_ int sd_bus_pending_method_calls(sd_bus *bus) {
+
+        /* Returns the number of currently pending asynchronous method calls. This is graceful, i.e. an
+         * unallocate (i.e. NULL) bus connection has no method calls pending. */
```
Comment (bluca) [inline src/libsystemd/sd-bus/sd-bus.c]: "unallocated"

File: src/libsystemd/sd-bus/sd-bus.c [hunk 2]
```diff
@@ -4466,3 +4466,22 @@ _public_ int sd_bus_enqueue_for_read(sd_bus *bus, sd_bus_message *m) {
...
+
+        if (!bus)
+                return 0;
+
+        assert_return(bus, -EINVAL);
```
Comment (bluca) [inline src/libsystemd/sd-bus/sd-bus.c]: this is redundant with the check above

File: man/sd_bus_pending_method_calls.xml
Comment (bluca) [inline man/sd_bus_pending_method_calls.xml:1]: version info is missing

Comment (bluca) [reply man/sd_bus_pending_method_calls.xml:1]: ah no it's a the bottom, nvm

**Review by bluca — APPROVED** [review]

File: man/sd_bus_pending_method_calls.xml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+  <refmeta>
+    <refentrytitle>sd_bus_pending_method_calls</refentrytitle>
+    <manvolnum>3</manvolnum>
+  </refmeta>
+
+  <refnamediv>
+    <refname>sd_bus_pending_method_calls</refname>
+
+    <refpurpose>Return number of currently pending, outgoing method calls</refpurpose>
```
Comment (YHNdnzj) [inline man/sd_bus_pending_method_calls.xml]:
```suggestion
    <refpurpose>Return the number of currently pending, outgoing method calls</refpurpose>
```

**Review by YHNdnzj — APPROVED** [review]
Please mention the new API in NEWS.

**Review by yuwata — APPROVED** [review]

**Comment by poettering** [comment]
i guess we should do this for v257. While the current behaviour of exiting on idle even if polkit authentication is still pending is mildly buggy it also doesn't really matter, the time window should be too small.

Moreover this probably needs fixing in more than just hostnamed, hence this is kinda incomplete.

**Pull request merged**